cmd/util/flags/cliflags/repo.go

```diff
@@ -0,0 +1,57 @@
+package cliflags
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/rs/zerolog/log"
+)
+
+const defaultBacalhauDir = ".bacalhau"
+
+// getDefaultRepo determines the appropriate default directory for storing repository data.
+// Priority order:
+// 1. If the environment variable BACALHAU_DIR is set and non-empty, use it.
+// 2. If the environment variable FIL_WALLET_ADDRESS is set, it tries to use ROOT_DIR.
+// 3. User's home directory with .bacalhau appended.
+// 4. User-specific configuration directory with .bacalhau appended.
+// 5. If all above fail, use .bacalhau in the current directory.
+// The function logs any errors encountered during the process and always returns a usable path.
+func getDefaultRepo() string {
+	if repoDir, set := os.LookupEnv("BACALHAU_DIR"); set && repoDir != "" {
+		return repoDir
+	}
+
+	if userHome, err := os.UserHomeDir(); err == nil {
+		return filepath.Join(userHome, defaultBacalhauDir)
+	}
+
+	if userDir, err := os.UserConfigDir(); err == nil {
+		return filepath.Join(userDir, defaultBacalhauDir)
+	}
```

Comment:
We discussed having the `UserConfigDir` as a default for the config path, and not for the repo path. Both locations should be decoupled from other. The only reason we are still considering `<repo>/config.yaml` as the default location for configs is for backward compatibility only. After implementing migration logic where we migrate configs from repo to `UserConfigDir`, then we should no longer look for configs under the repo

Our tenant in this area should be less-is-more. We should avoid doing more magic on-behalf of the user and avoid having too many paths and forks in the logic of config setup. My original preference was to even not have a default config path, but I understand that will complicate migration logic